packages/harpparts/src/apparatus/utils/is-hole-valid/is-hole-valid.test.ts

```diff
@@ -0,0 +1,346 @@
+    drawbends: [1, 2, 3, 4, 5, 6],
+    overblows: [],
+    overdraws: [],
+    valvedblows: [],
+    valveddraws: [],
+  }
+  expect(isHoleValid(hole)).toStrictEqual([HoleErrors.TooManyBends])
+})
+
+test('isHoleValid returns no errors when there are 5 blowbends on a hole', () => {
+  const hole: Hole = {
+    blow: 6,
+    draw: 0,
+    blowbends: [1, 2, 3, 4, 5],
+    drawbends: [],
+    overblows: [],
+    overdraws: [],
+    valvedblows: [],
+    valveddraws: [],
+  }
+  expect(isHoleValid(hole)).toStrictEqual([])
+})
+
+test('isHoleValid returns an error when there are more than 5 blowbends on a hole', () => {
+  const hole: Hole = {
+    blow: 7,
+    draw: 0,
+    blowbends: [1, 2, 3, 4, 5, 6],
+    drawbends: [],
+    overblows: [],
+    overdraws: [],
+    valvedblows: [],
+    valveddraws: [],
+  }
+  expect(isHoleValid(hole)).toStrictEqual([HoleErrors.TooManyBlowbends])
+})
+
+test('isHoleValid returns no errors when there are 2 overblows on a hole', () => {
+  const hole: Hole = {
+    blow: 0,
+    draw: 2,
+    blowbends: [],
+    drawbends: [],
+    overblows: [3, 4],
+    overdraws: [],
+    valvedblows: [],
+    valveddraws: [],
+  }
+  expect(isHoleValid(hole)).toStrictEqual([])
+})
+
+test('isHoleValid returns an error when there are more than 2 overblows on a hole', () => {
+  const hole: Hole = {
+    blow: 0,
+    draw: 2,
+    blowbends: [],
+    drawbends: [],
+    overblows: [3, 4, 5],
+    overdraws: [],
+    valvedblows: [],
+    valveddraws: [],
+  }
+  expect(isHoleValid(hole)).toStrictEqual([HoleErrors.TooManyOverblows])
+})
+
+test('isHoleValid returns no errors when there are 2 overdraws on a hole', () => {
+  const hole: Hole = {
+    blow: 2,
+    draw: 0,
+    blowbends: [],
+    drawbends: [],
+    overblows: [],
+    overdraws: [3, 4],
+    valvedblows: [],
+    valveddraws: [],
+  }
+  expect(isHoleValid(hole)).toStrictEqual([])
+})
+
+test('isHoleValid returns an error when there are more than 2 overdraws on a hole', () => {
+  const hole: Hole = {
+    blow: 2,
+    draw: 0,
+    blowbends: [],
+    drawbends: [],
+    overblows: [],
+    overdraws: [3, 4, 5],
+    valvedblows: [],
+    valveddraws: [],
+  }
+  expect(isHoleValid(hole)).toStrictEqual([HoleErrors.TooManyOverdraws])
+})
+
+test('isHoleValid returns no errors when all the bends are consecutively ascending', () => {
+  const hole: Hole = {
+    blow: 0,
+    draw: 4,
+    blowbends: [],
+    drawbends: [1, 2, 3],
+    overblows: [],
+    overdraws: [],
+    valvedblows: [],
+    valveddraws: [],
+  }
+  expect(isHoleValid(hole)).toStrictEqual([])
+})
+
+test('isHoleValid returns an error when the bends are not consecutively ascending', () => {
+  const hole: Hole = {
+    blow: 0,
+    draw: 4,
+    blowbends: [],
+    drawbends: [],
+    overblows: [],
+    overdraws: [],
+    valvedblows: [],
+    valveddraws: [],
+  }
+  expect(isHoleValid({ ...hole, drawbends: [2, 1, 0] })).toStrictEqual([
+    HoleErrors.NonconsecutiveBends,
```

Comment:
The names and capitalisations in the names of `HoleErrors` enum need to be reviewed.

Reply:
Done